File: src/main/java/com/thunderbear06/ai/task/tasks/InteractEntityTask.java
package com.thunderbear06.ai.task.tasks;

import com.thunderbear06.entity.android.AndroidEntity;
import net.minecraft.world.InteractionHand;
import net.minecraft.world.entity.LivingEntity;

public class InteractEntityTask extends MoveToEntityTask {
    private boolean complete;

    public InteractEntityTask(AndroidEntity android, double moveSpeed, LivingEntity entity) {
        super(android, moveSpeed, entity);
    }

    @Override
    public String getName() {
        return "usingEntity";
    }

    @Override
    public boolean shouldTick() {
        return getTarget().isAlive() && !complete;
    }

    @Override
    public void tick() {
        if (isInRange(2.0D)) {
            android.getLookControl().setLookAt(getTarget());
            android.brain.getModules().interactionModule.interactWithEntity(InteractionHand.MAIN_HAND, getTarget());
            complete = true;
        } else {
            super.tick();
        }
    }
}

File: src/main/kotlin/com/thunderbear06/ai/task/tasks/InteractEntityTask.kt
package com.thunderbear06.ai.task.tasks

import com.thunderbear06.entity.android.AndroidEntity
import net.minecraft.world.InteractionHand
import net.minecraft.world.entity.LivingEntity

class InteractEntityTask(android: AndroidEntity, moveSpeed: Double, entity: LivingEntity) : MoveToEntityTask(android, moveSpeed, entity) {
    private var complete = false

    override val name: String = "usingEntity"

    override fun shouldTick(): Boolean = target.isAlive && !complete

    override fun tick() {
        if (isInRange(2.0)) {
            android.lookControl.setLookAt(target)
            android.brain.modules.interactionModule.interactWithEntity(InteractionHand.MAIN_HAND, target)
            complete = true
        } else {
            super.tick()
        }
    }
}
